letsencrypt/plugins/webroot.py

```diff
@@ -95,18 +110,18 @@ def _path_for_achall(self, achall):
 
     def _perform_single(self, achall):
         response, validation = achall.response_and_validation()
+
         path = self._path_for_achall(achall)
         logger.debug("Attempting to save validation to %s", path)
-        with open(path, "w") as validation_file:
-            validation_file.write(validation.encode())
-
-        # Set permissions as parent directory (GH #1389)
-        parent_path = self.full_roots[achall.domain]
-        stat_parent_path = os.stat(parent_path)
-        filemode = stat.S_IMODE(stat_parent_path.st_mode)
-        # Remove execution bit (not needed for this file)
-        os.chmod(path, filemode & ~stat.S_IEXEC)
-        os.chown(path, stat_parent_path.st_uid, stat_parent_path.st_gid)
+
+        old_umask = os.umask(0o022)
+
+        try:
+            with open(path, "w") as validation_file:
+                # Change permissions to be world-readable, owner-writable (GH #1795)
```

Comment:
I think the comment about the permissions should be moved to the `os.umask` call.